src/motion-queue.h

```diff
@@ -72,9 +72,23 @@ struct MotionSegment {
 #endif
 } __attribute__((packed));
 
+//Define a struct to easily access the status register
+struct QueueStatus{
+    uint32_t counter : 24;
+    uint32_t index : 8;
+}__attribute__((packed));
+
+
 // Low level motion queue operations.
 class MotionQueue {
 public:
+    class EventReceiver {
+    public:
+      virtual void update_absolute_position() = 0; //callback to update the absolute position
```

Comment:
This does not really explain what is happening. The name implies that this receives some kind of Event. And apparently that is update_absolute_position(). But wouldn't we expect with then that this contains the new information in the parameter or something ?

Is 'EventReceiver' a good name ? Can we be more specific (PositionChangeCalback or something ?). (Haven't looked at the rest of the code yet, so I might mis-understand. But I think things should be immediately understandable for someone reading the header file).